021-special-repository-for-st-clients-based-on-example-clients.md

```diff
@@ -0,0 +1,71 @@
+# Special repository for ST clients based on example clients
+
+This proposal suggests creating a new repository for `Strimzi` ST client based on [Strimzi client-examples](https://github.com/strimzi/client-examples).
+
+## Current situation
+
+Currently, we are using two clients in our STs:
+ - `InternalKafkaClients` (based on `test-client` image) 
+ - `example clients` (from [Strimzi client-examples](https://github.com/strimzi/client-examples)).
+
+The plan is to remove the `InternalKafkaClients` and keep only `example clients`. 
+The `test-client` is not sufficient anymore, it can create single producer/consumer, send/receive messages, and then we can assert result.
+We are stuck here, and we need to wait until a producer/consumer is finished. 
+With `example clients` we are able to do a lot more.
+For example - create a continuous job for sending messages with delay, 
+_stack_ the producers to create a _traffic_,  add extra configuration, use different types of producer/consumer (for Bridge, Kafka, ...) and many more.  
+
+## Motivation
+
+While testing `Strimzi` we need, in some cases, special configuration of clients, which is not implemented in the `client-examples`.
+As we discussed offline, the `client-examples` repository should be really _exemplary_, 
```

Comment:
I don't like to see mention of offline discussion: It suggests things are happening in private, for unknown reasons. I understand that sometimes offline discussion just happens in a non-public forum by accident. In those cases the discussion and outcome should be summarised publicly somewhere for anyone else to contribute. 

In this case I've no idea who "we" means, and in any case I don't think it matters: It is enough to say that client-examples should be exemplary, and that a dedicated test client is better than pushing the example clients in a direction that makes them less useful as examples.

Reply:
Good point, thanks! :)